src/zjs_net.c

```diff
@@ -0,0 +1,480 @@
+    int i;
+    for (i = 0; i < argc; ++i) {
+        if (jerry_value_is_number(argv[i])) {
+            port = jerry_get_number_value(argv[i]);
+        } else if (jerry_value_is_string(argv[i])) {
+            zjs_copy_jstring(argv[i], hostname, &size);
+            if (!size) {
+                return zjs_error("invalid arguments");
+            }
+        } else if (jerry_value_is_function(argv[i])) {
+            zjs_add_event_listener(this, "listening", argv[i]);
+        }
+    }
+
+    net_handle_t *handle = NULL;
+
+    jerry_get_object_native_handle(this, (uintptr_t*)&handle);
+
+    struct sockaddr_in6 my_addr6 = { 0 };
+
+    my_addr6.sin6_family = AF_INET6;
+    my_addr6.sin6_port = htons(port);
+
+    CHECK(net_context_bind(handle->tcp_sock,
+                           (struct sockaddr *)&my_addr6,
+                           sizeof(struct sockaddr_in6)));
+    CHECK(net_context_listen(handle->tcp_sock, 0));
+
+    handle->listening = 1;
+    handle->port = port;
+    memcpy(&handle->local, &my_addr6, sizeof(struct sockaddr *));
+    zjs_obj_add_boolean(this, true, "listening");
+
+    zjs_trigger_event(this, "listening", NULL, 0, NULL, NULL);
+
+    CHECK(net_context_accept(handle->tcp_sock, tcp_accepted, 0, handle));
+
+    DBG_PRINT("listening for connection to %s:%u\n", hostname, port);
+
+    return ZJS_UNDEFINED;
+}
+
+
+static jerry_value_t zjs_net_create_server(const jerry_value_t function_obj,
+                                           const jerry_value_t this,
+                                           const jerry_value_t argv[],
+                                           const jerry_length_t argc)
+{
+    int ret;
+    jerry_value_t connection_cb = 0;
+    if (argc > 0) {
+        if (!jerry_value_is_function(argv[0])) {
+            return zjs_error("invalid arguments");
+        } else {
+            connection_cb = argv[0];
+        }
+    }
+    jerry_value_t server = jerry_create_object();
+
+    zjs_obj_add_boolean(server, false, "listening");
+    zjs_obj_add_number(server, NET_DEFAULT_MAX_CONNECTIONS, "maxConnections");
+
+    zjs_obj_add_function(server, server_listen, "listen");
+    zjs_obj_add_function(server, server_close, "close");
+    zjs_obj_add_function(server, server_get_connections, "getConnections");
+
+
+    zjs_make_event(server, ZJS_UNDEFINED);
+
+    if (connection_cb) {
+        zjs_add_event_listener(server, "connection", connection_cb);
+    }
+
+    net_handle_t *handle = zjs_malloc(sizeof(net_handle_t));
+
+    CHECK(net_context_get(AF_INET6, SOCK_STREAM, IPPROTO_TCP,
+            &handle->tcp_sock))
+
+    jerry_set_object_native_handle(server, (uintptr_t)handle, NULL);
+
+    handle->server = server;
+    handle->listening = 0;
+
+    DBG_PRINT("creating server: context=%p\n", handle->tcp_sock);
+
+    return server;
+}
+
+jerry_value_t zjs_net_init()
+{
+#if defined(CONFIG_NET_L2_BLUETOOTH)
+    if (bt_enable(NULL)) {
+        NET_ERR("Bluetooth init failed");
```

Comment:
use ERR_PRINT?  does NET_ERR gets printed? or unless you enable some config flags only?